src/tsdb.c

```diff
@@ -206,6 +206,34 @@ void RenameSeriesFrom(RedisModuleCtx *ctx, RedisModuleString *key) {
     renameFromKey = key;
 }
 
+void RestoreKey(RedisModuleCtx *ctx, RedisModuleString *keyname) {
+    Series *series;
+    RedisModuleKey *key = NULL;
+    SilentGetSeries(ctx, keyname, &key, &series, REDISMODULE_READ);
+
+    CompactionRule *rule = series->rules;
+    while (rule != NULL) {
+        RedisModuleKey *destKey = NULL;
+        Series *destSeries;
+        const int status =
+            SilentGetSeries(ctx, rule->destKey, &destKey, &destSeries, REDISMODULE_WRITE);
+        if (status != TRUE) {
+            continue;
+        }
+
+        RedisModule_RetainString(ctx, keyname);
+        destSeries->srcKey = keyname;
+        if (destKey) {
+            RedisModule_CloseKey(destKey);
+        }
+        rule = rule->nextRule;
+    }
+
+    if (key) {
```

Comment:
@danni-m when can a key be NULL? if it's NULL I guess it can only happen if it fails to open the key on line 212, then shouldn't you check the status on 212?

Reply:
@danni-m why do you need the NULL check?

Reply:
Just another precaution, I checked the redis codebase and its redundant as it also checked by `RedisModule_CloseKey`.